// paystack-sdk/dist/transaction/transaction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Transaction = void 0;
/**
 * # Tansactions
 * The transaction API allows you create and manage
 * payments on your integration
 */
class Transaction {
    constructor(http) {
        this.http = http;
    }
    /**
     * Initialize a transaction
     * @param {InitializeTransaction} data **Body Param**
     */
    async initialize(data) {
        return await this.http.post('/transaction/initialize', JSON.stringify(data));
    }
    async verify(reference) {
        return await this.http.get(`/transaction/verify/${reference}`);
    }
    async list(queryParams) {
        return await this.http.get('/transaction', {
            params: { ...queryParams },
        });
    }
    async fetch(id) {
        return await this.http.get(`/transaction/:${id}`);
    }
    async chargeAuthorization(data) {
        return await this.http.post('/transaction/charge_authorization', JSON.stringify(data));
    }
    async checkAuthorization(data) {
        return await this.http.post('/transaction/check_authorization', JSON.stringify(data));
    }
    async viewTimeline(id) {
        return await this.http.get(`/transaction/timeline/${id}`);
    }
    async total(queryParams) {
        return await this.http.get('/transaction/totals', {
            params: { ...queryParams },
        });
    }
    async export(queryParams) {
        return await this.http.get('/transaction/export', {
            params: { ...queryParams },
        });
    }
    async partialDebit(data) {
        return await this.http.post('/transaction/partial_debit', JSON.stringify(data));
    }
}
exports.Transaction = Transaction;
